pysmt/test/test_back.py

```diff
@@ -63,6 +63,15 @@ def do_back(self, solver_name, z3_string_buffer=False):
                     # with the SMT-LIB. We might consider extending our
                     # parser.
                     continue
+
+                # The back conversion via SMTLIB of Z3 has a bug
```

Comment:
The issue seems to have been resolved on Z3 side. Can you check? If so, we might consider upgrading.

Reply:
I upgraded to Z3 4.8.5, but the solver prints the empty-string symbol as 'null' (but at least it does not crash). I implemented a simple rewriting as for MathSAT.